src/js/components/devices/terminal.js

```diff
@@ -118,6 +131,19 @@ export const Terminal = props => {
           }
         } else if (obj.hdr.typ === MessageTypeShell) {
           term.write(byteArrayToString(obj.body));
+        } else if (obj.hdr.typ == MessageTypePing) {
+          if (healthcheckTimeout) {
+            clearTimeout(healthcheckTimeout);
+          }
+          const proto_header = { proto: 1, typ: MessageTypePong, sid: sessionId, props: null };
+          const msg = { hdr: proto_header, body: null };
+          const encodedData = MessagePack.encode(msg);
+          socket.send(encodedData);
+          //
+          var timeout = parseInt((obj.hdr.props || {}).timeout);
```

Comment:
In what case would the timeout not be set? should the connection maybe still set a timeout in that case? I'm not steeped in the protocol details or the entire interactions between the device/ backend/ gui - so I'll appreciate any details possible!

Reply:
If the mender-connect disables it; it is not supported as of today (the interval is hard-coded), but who knows in the future... in this case, the UI still replies to "ping" requests with a pong, but does not set any client-side timeout. To be honest, the important thing is being sure the user (UI) is still around, the other way around is a "nice to have" feature.